src/util/expandThoughts.js

```diff
@@ -32,13 +32,15 @@ const publish = new URLSearchParams(window.location.search).get('publish') != nu
     ...
   }
 */
-export const expandThoughts = (path, thoughtIndex, contextIndex, contextViews = {}, contextChain = [], { depth = 0 } = {}) => {
+export const expandThoughts = (path, thoughtIndex, contextIndex, contextViews = {}, contextChain = [], { depth = 0, distanceFromTableAncestor = -1 } = {}) => {
 
   if (
     // arbitrarily limit depth to prevent infinite context view expansion (i.e. cycles)
     depth > MAX_EXPAND_DEPTH ||
     // suppress expansion if left command is held down after cursorDown
-    globals.suppressExpansion
+    globals.suppressExpansion ||
+    // if distance from the laest ancestor with table view is more than 1 then it should stop expanding
+    distanceFromTableAncestor > 1
```

Comment:
This will break the `=pin` functionality. Instead of short circuiting, use `isTableColumn2` to affect the main conditional (L71). In particular, you will want to modify `isOnlyChildNoUrl` with `&& !isTableColumn2`.